core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java

```diff
@@ -455,6 +455,51 @@ public boolean accept(CarbonFile pathName) {
     return null;
   }
 
+  public Map<String, List<CarbonFile>> getDeleteDeltaFilesForSegment(final Segment seg) {
+    String segmentPath = CarbonTablePath.getSegmentPath(
+      identifier.getTablePath(), seg.getSegmentNo());
```

Comment:
if SegmentUpdateDetails  donot contains seg, we shall return empty result directly.
which can save a lot of IO overhead

Reply:
Done.